File: Geometry/Solutions/Largest_Square_Area_Inside_Two_Rectangles.py
class Solution:
    def largestSquareArea(self, bottomLeft: List[List[int]], topRight: List[List[int]]) -> int:
        n = len(bottomLeft)
        max_area = 0

        # Iterate over all pairs of rectangles
        for i in range(n):
            for j in range(i + 1, n):

                # Find intersection bottom-left coordinates
                x1 = max(bottomLeft[i][0], bottomLeft[j][0])
                y1 = max(bottomLeft[i][1], bottomLeft[j][1])

                # Find intersection top-right coordinates
                x2 = min(topRight[i][0], topRight[j][0])
                y2 = min(topRight[i][1], topRight[j][1])

                # If rectangles do not overlap, skip this pair
                if x2 <= x1 or y2 <= y1:
                    continue

                # Width and height of the intersection rectangle
                width = x2 - x1
                height = y2 - y1

                # Largest square side that can fit in the intersection
                side = min(width, height)

                # Update maximum square area
                max_area = max(max_area, side * side)

        return max_area

File: Geometry/Solutions/Largest_Square_Area_Inside_Two_Rectangles.cpp
class Solution {
public:
    int largestSquareArea(vector<vector<int>>& bottomLeft,
                          vector<vector<int>>& topRight) {

        int n = bottomLeft.size();
        long long maxArea = 0;

        // Check every pair of rectangles
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {

                // Intersection bottom-left corner
                int x1 = max(bottomLeft[i][0], bottomLeft[j][0]);
                int y1 = max(bottomLeft[i][1], bottomLeft[j][1]);

                // Intersection top-right corner
                int x2 = min(topRight[i][0], topRight[j][0]);
                int y2 = min(topRight[i][1], topRight[j][1]);

                // If no overlap, skip
                if (x2 <= x1 || y2 <= y1)
                    continue;

                // Dimensions of intersection rectangle
                int width  = x2 - x1;
                int height = y2 - y1;

                // Largest square inside this intersection
                int side = min(width, height);

                // Update maximum area
                maxArea = max(maxArea, 1LL * side * side);
            }
        }

        return (int)maxArea;
    }
};